src/java.base/share/classes/sun/security/ssl/CertificateAuthoritiesExtension.java

```diff
@@ -277,7 +284,11 @@ public void consume(ConnectionContext context,
                     new CertificateAuthoritiesSpec(shc, buffer);
 
             // Update the context.
-            shc.peerSupportedAuthorities = spec.getAuthorities();
+            try {
+                shc.peerSupportedAuthorities = spec.getAuthorities();
+            } catch (IllegalArgumentException iae) {
+                shc.conContext.fatal(Alert.DECODE_ERROR, "X500Principal could not be parsed", iae);
```

Comment:
In the context, it may be easier to catch if the message is about the authorities, and easier to update getAuthorities() implementation if needed in the future.
- "X500Principal could not be parsed"
+ "Peer authorities could not be parsed"